### PR TITLE
feat: adds update stronghold logic

### PR DESCRIPTION
**Pull request by jeeanribeiro**
## Summary

Same in https://github.com/iotaledger/firefly/pull/6123 just with a different base branch

### Platforms

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [x] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

## Checklist

-   [x] I have followed the contribution guidelines for this project
-   [x] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [x] I have added or modified tests that prove my changes work as intended
-   [x] I have verified that new and existing unit tests pass locally with my changes
-   [x] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by maxwellmattryan — APPROVED** [review]
Code looks good and it's working :smiling_face_with_tear:

File: packages/desktop/electron/main.js [hunk 1]
```diff
@@ -423,6 +424,15 @@ ipcMain.handle('check-if-file-exists', (_e, filePath) => {
     return fs.existsSync(`${directory}/__storage__/${filePath}`)
 })
 
+function ensureDirectoryExistence(filePath) {
```
Comment (nicole-obrien) [inline packages/desktop/electron/main.js:427]: Perhaps if we are not using this in the PR we can add it as a PR onto develop if it is an improvement we think its worth having

File: packages/desktop/views/onboarding/views/CongratulationsView.svelte [hunk 2]
```diff
@@ -24,7 +23,7 @@
             checkOrConnectLedger(_continue)
         } else {
             updateOnboardingProfile({
-                strongholdVersion: features.onboarding.strongholdVersionCheck.enabled ? STRONGHOLD_VERSION : undefined,
+                strongholdVersion: STRONGHOLD_VERSION,
```
Comment (nicole-obrien) [inline packages/desktop/views/onboarding/views/CongratulationsView.svelte:26]: This is fine here, but would make more sense to be on the build initial onboarding profile object

File: packages/desktop/views/onboarding/views/profile-recovery/views/SuccessView.svelte [hunk 3]
```diff
@@ -30,7 +30,7 @@
 
 <OnboardingLayout {onBackClick}>
     <div slot="leftpane__content">
-        <div class="flex flex-col items-center bg-gray-100 dark:bg-gray-900 rounded-2xl mt-10 p-5 pt-10 relative">
+        <div class="flex flex-col items-center bg-gray-100 dark:bg-gray-900 rounded-2xl p-5 pt-10 relative">
```
Comment (nicole-obrien) [inline packages/desktop/views/onboarding/views/profile-recovery/views/SuccessView.svelte:33]: Needed?

File: packages/desktop/electron/main.js (hunk 1 above)
Comment (jeeanribeiro) [reply packages/desktop/electron/main.js:427]: Nice idea, added here: https://github.com/iotaledger/firefly/pull/6652/files

**Comment by nicole-obrien** [comment]
Closing so we review it once into develop and create a new release branch

**Pull request closed without merging**